core/src/main/java/io/dekorate/utils/TopologicalSort.java

```diff
@@ -32,7 +33,8 @@ private TopologicalSort() {
    */
   public static List<Decorator> sortDecorators(Collection<Decorator> items) {
     Collection<Node> nodes = adaptToNodes(items);
-    return sortNodes(nodes);
+    // Order the nodes by the decorator class to guarantee the initial order.
+    return sortNodes(new TreeSet<>(nodes));
```

Comment:
It's not clear to me what you mean by initial order.

Reply:
The idea is to get a determinist order if we always get the same input. For example, if our input is a set of three decorators with no dependencies on each other, the final result order of these three decorators might vary in different executions.

The idea I had to address this situation is to introduce an initial order based on the decorator class name (it can be any other).

Reply:
> The idea is to get a determinist order if we always get the same input. For example, if our input is a set of three decorators with no dependencies on each other, the final result order of these three decorators might vary in different executions.

The order of decorators should not be important to us as long as they are topologically sorted. Even if it's possible the get different results that both satisfy the constraints, the end result (the generated json/yaml files) should be deterministic. If the end result is not deterministic, then we probably have not properly expressed those constraints and we should focus on that.

TDLR: I don't mind having the decorators in a predictable order however, we shouldn't have to,  are we sure we are dealing with the issue at the right level?